lock/client/client.go

```diff
@@ -0,0 +1,136 @@
+// Package client implements a client to the volmaster to acquire locks.
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/contiv/volplugin/config"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Driver is the main force behind the lock module, it controls all methods and
+// several variables.
+type Driver struct {
+	master         string
+	mountStopChans map[string]chan struct{}
+	mountMutex     sync.Mutex
```

Comment:
minor: just wondering, if we shall just embed the `sync.Mutex` directly here? That way we can just call `d.Lock()` and `d.Unlock()` every, than more verbose `d.mountMutex...`